src/py/flwr/client/secure_aggregation/secaggplus_handler.py

```diff
@@ -0,0 +1,453 @@
+        shared_key = state.ss2_dict[src]
+        plaintext = decrypt(shared_key, ciphertext)
+        _src, dst, b_share, sk1_share = share_keys_plaintext_separate(plaintext)
+        available_clients.append(src)
+        if src != _src:
+            raise ValueError(
+                f"Client {state.sid}: received ciphertext from {_src} instead of {src}"
+            )
+        if dst != state.sid:
+            ValueError(
+                f"Client {state.sid}: received an encrypted message"
+                f"for Client {dst} from Client {src}"
+            )
+        state.rd_seed_share_dict[src] = b_share
+        state.sk1_share_dict[src] = sk1_share
+
+    # fit client
+    parameters_bytes = cast(List[bytes], named_values["parameters"])
+    parameters = [bytes_to_ndarray(w) for w in parameters_bytes]
+    if isinstance(state.client, Client):
+        fit_res = state.client.fit(
+            FitIns(parameters=ndarrays_to_parameters(parameters), config={})
+        )
+        parameters_factor = fit_res.num_examples
+        parameters = parameters_to_ndarrays(fit_res.parameters)
+    elif isinstance(state.client, NumPyClient):
+        parameters, parameters_factor, _ = state.client.fit(parameters, {})
+    else:
+        log(ERROR, "Client %d: fit function is none", state.sid)
+
+    # Quantize weight update vector
+    quantized_parameters = quantize(
+        parameters, state.clipping_range, state.target_range
+    )
+
+    quantized_parameters = parameters_multiply(quantized_parameters, parameters_factor)
+    quantized_parameters = factor_combine(parameters_factor, quantized_parameters)
+
+    dimensions_list: List[Tuple[int, ...]] = [a.shape for a in quantized_parameters]
+
+    # add private mask
+    private_mask = pseudo_rand_gen(state.rd_seed, state.mod_range, dimensions_list)
+    quantized_parameters = parameters_addition(quantized_parameters, private_mask)
+
+    for client_id in available_clients:
+        # add pairwise mask
+        shared_key = generate_shared_key(
+            bytes_to_private_key(state.sk1),
+            bytes_to_public_key(state.public_keys_dict[client_id][0]),
+        )
+        pairwise_mask = pseudo_rand_gen(shared_key, state.mod_range, dimensions_list)
+        if state.sid > client_id:
+            quantized_parameters = parameters_addition(
+                quantized_parameters, pairwise_mask
+            )
+        else:
+            quantized_parameters = parameters_subtraction(
+                quantized_parameters, pairwise_mask
+            )
+
+    # Take mod of final weight update vector and return to server
+    quantized_parameters = parameters_mod(quantized_parameters, state.mod_range)
+    # return ndarrays_to_parameters(quantized_parameters)
+    log(INFO, "Client %d: stage 2 completes. uploading masked parameters...", state.sid)
+    return {
+        "masked_parameters": [ndarray_to_bytes(arr) for arr in quantized_parameters]
+    }
+
+
+def _unmasking(
```

Comment:
Part of the renaming suggestion above:
```suggestion
def _unmask(
```